client/go/outline/electron/outline_device.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2024 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"log/slog"
+
+	"github.com/Jigsaw-Code/outline-apps/client/go/outline"
+	perrs "github.com/Jigsaw-Code/outline-apps/client/go/outline/platerrors"
+	"github.com/Jigsaw-Code/outline-sdk/network"
+	"github.com/Jigsaw-Code/outline-sdk/network/dnstruncate"
+	"github.com/Jigsaw-Code/outline-sdk/network/lwip2transport"
+)
+
+type outlineDevice struct {
+	network.IPDevice
+
+	c        *outline.Client
+	pkt      network.DelegatePacketProxy
+	routeUDP *bool
+
+	remote, fallback network.PacketProxy
+}
+
+func (d *outlineDevice) Connect() (perr *perrs.PlatformError) {
+	var err error
+
+	d.remote, err = network.NewPacketProxyFromPacketListener(d.c.PacketListener)
+	if err != nil {
+		return errSetupHandler("failed to create datagram handler", err)
+	}
+	slog.Debug("[OutlineNetDev] remote UDP handler created")
+
+	if d.fallback, err = dnstruncate.NewPacketProxy(); err != nil {
```

Comment:
Not for this PR, but dnstruncate is actually a bit fragile. If a single connectivity query fails, we shut down UDP completely.

I think a better approach would be to continue to send UDP packets all the time, and only change the behavior for DNS specifically.